include/zephyr/posix/signal.h

```diff
@@ -12,6 +12,74 @@
 extern "C" {
 #endif
 
+#ifdef CONFIG_POSIX_SIGNAL
+BUILD_ASSERT(CONFIG_POSIX_SIGNAL_NSIG > CONFIG_POSIX_SIGNAL_SIGRTMIN);
+
+#define _NSIG     CONFIG_POSIX_SIGNAL_NSIG
+
+#define SIGHUP    1
```

Comment:
It would be nice to re-use `scripts/build/gen_strerror_table.py` a bit the same way it's used in `lib/libc/minimal/CMakeLists.txt`.

In this file, e.g.
```cpp
#define SIGHUP 1 /* Hangup */
```

Reply:
most of the differences between `strerror` & `strsignal` can be solved by adding more options to the script, but I'm not sure what to do about the preprocessor defines of `strerror` so I've opted to create another script for `strsignal` instead of modifying `gen_strerror_table.py`, let me know what you  think.

Reply:
Would be nice to eventually merge the two, but no pressure yet.